test/settings_test.dart

```diff
@@ -135,5 +149,170 @@ void main() {
       expect(find.text('Settings'), findsOneWidget);
       expect(find.text('Auto'), findsOneWidget);
     });
+
+    testWidgets('Ask Permissions - normal scenario',
+        (WidgetTester tester) async {
+      when(mockAuthProvider.currentUser).thenAnswer((_) =>
```

Comment:
If you group the tests, you can move this line to the `setUp` of that group.

https://stackoverflow.com/questions/33017366/dart-how-run-a-function-after-or-before-each-test